python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -679,40 +680,37 @@ def can_reconcile_candidate(candidate: AssetKeyPartitionKey) -> bool:
         can_reconcile_fn=can_reconcile_candidate,
     )
 
-    def get_unresolved_parent_asset_keys(candidate: AssetKeyPartitionKey) -> AbstractSet[AssetKey]:
-        """Returns the set of parent asset keys which would be unresolved if this candidate were
-        materialize this tick.
+    def get_waiting_on_asset_keys(candidate: AssetKeyPartitionKey) -> FrozenSet[AssetKey]:
+        """Returns the set of ancestor asset keys that must be materialized before this asset can be
+        materialized.
         """
         from dagster._core.definitions.external_asset_graph import ExternalAssetGraph
 
-        return {
-            parent.asset_key
-            for parent in asset_graph.get_parents_partitions(
-                instance_queryer,
-                instance_queryer.evaluation_time,
-                candidate.asset_key,
-                candidate.partition_key,
-            ).parent_partitions
-            if (
-                # parent will not be materialized this tick
-                not (
-                    _will_materialize_for_conditions(conditions_by_asset_partition.get(parent))
-                    # the parent must have the same partitioning / partition key to be materialized
-                    # alongside the candidate
-                    and asset_graph.have_same_partitioning(parent.asset_key, candidate.asset_key)
-                    and parent.partition_key == candidate.partition_key
-                    # the parent must be in the same repository to be materialized alongside the candidate
-                    and (
-                        not isinstance(asset_graph, ExternalAssetGraph)
-                        or asset_graph.get_repository_handle(candidate.asset_key)
-                        == asset_graph.get_repository_handle(parent.asset_key)
-                    )
+        unresolved_parents = set()
```

Comment:
This should be "ancestors", not parents, right? Also, would `unreconciled` make sense instead of `unresolved`? Wary of introducing a new term here.

Reply:
ah yes, this was me trying to play around with the potential new terminology but putting it back.